src/components/SwappingCarbonOffset.tsx

```diff
@@ -0,0 +1,928 @@
+import { ArrowRightOutlined, LoadingOutlined, PauseCircleOutlined } from '@ant-design/icons'
+import { Web3Provider } from '@ethersproject/providers'
+import { ChainId, Execution, ExecutionSettings, Process, StepTool } from '@lifinance/sdk'
+import { getRpcProvider } from '@lifinance/sdk/dist/connectors'
+import { useWeb3React } from '@web3-react/core'
+import {
+  Avatar,
+  Button,
+  Divider,
+  Modal,
+  Row,
+  Space,
+  Spin,
+  Timeline,
+  Tooltip,
+  Typography,
+} from 'antd'
+import BigNumber from 'bignumber.js'
+import { constants, ethers } from 'ethers'
+import { GatewayBatchStates, Sdk } from 'etherspot'
+import { useEffect, useState } from 'react'
+import { useMediaQuery } from 'react-responsive'
+
+import walletIcon from '../assets/wallet.png'
+import { KLIMA_CARBON_OFFSET_CONTRACT, TOUCAN_BCT_ADDRESS } from '../constants'
+import LiFi from '../LiFi'
+import {
+  getOffsetCarbonTransaction,
+  getSetAllowanceTransaction,
+} from '../services/ethersportTxService'
```

Comment:
Good old `ethersport`, the origin of all e-sports

Reply:
will fix the typo before merging